smac_planner/src/a_star.cpp

```diff
@@ -197,15 +192,12 @@ bool AStarAlgorithm::createPath(IndexPath & path, int & iterations, const float
 
     // 4) Expand neighbors of Nbest not visited
     neighbors.clear();
-    getNeighbors(current_node, neighbors);
+    current_node->getNeighbors(neighbors, _graph.get(), _traverse_unknown);
```

Comment:
I wonder if we should leave the getNeighbors outside of the Node implementation for now. The static thing I think should work fine but I wonder if we're changing too much at once that could effect performance.

Reply:
That's a valid concern. I will use the Willow Garage map to benchmark the templated implementation and additionally I can move the getNeighbors outside of Node and test that as well.